File: domain/src/main/java/mixter/domain/core/subscription/FollowerRepository.java
package mixter.domain.core.subscription;

import mixter.domain.identity.UserId;

import java.util.Set;

public interface FollowerRepository {
    Set<UserId> getFollowers(UserId authorId);

    void saveFollower(UserId followee, UserId follower);

    void removeFollower(UserId followee, UserId follower);
}

File: domain/src/main/scala/mixter/domain/subscription/FollowerRepository.scala
package mixter.domain.subscription

import mixter.domain.identity.UserId

trait FollowerRepository {

  def getFollowers(authorId: UserId): Set[UserId]

  def saveFollower(followee: UserId, follower: UserId): Unit

  def removeFollower(followee: UserId, follower: UserId): Unit
}
